target/linux/ramips/patches-6.6/831-mmc-mtk-sd-initialize-pad-delay-and-drive-strength.patch

```diff
@@ -0,0 +1,33 @@
+--- a/drivers/mmc/host/mtk-sd.c
++++ b/drivers/mmc/host/mtk-sd.c
+@@ -75,8 +75,12 @@
+ #define MSDC_PATCH_BIT   0xb0
+ #define MSDC_PATCH_BIT1  0xb4
+ #define MSDC_PATCH_BIT2  0xb8
++#define MSDC_PAD_CTRL0   0xe0
++#define MSDC_PAD_CTRL1   0xe4
++#define MSDC_PAD_CTRL2   0xe8
+ #define MSDC_PAD_TUNE    0xec
+ #define MSDC_PAD_TUNE0   0xf0
++#define MSDC_PAD_TUNE1   0xf4
+ #define PAD_DS_TUNE      0x188
+ #define PAD_CMD_TUNE     0x18c
+ #define EMMC51_CFG0	 0x204
+@@ -1795,6 +1799,17 @@ static void msdc_init_hw(struct msdc_hos
+ 				     MSDC_PAD_TUNE_RXDLYSEL);
+ 	}
+ 
++	/* Set pins drive strength */
++	writel(0x000d0044, host->base + MSDC_PAD_CTRL0);
++	writel(0x000e0044, host->base + MSDC_PAD_CTRL1);
++	writel(0x000e0044, host->base + MSDC_PAD_CTRL2);
++
++	/* Set pad delay */
++	sdr_clr_bits(host->base + MSDC_PAD_TUNE, 0xffdf1f1f);
++	sdr_set_bits(host->base + MSDC_PAD_TUNE, 0x84101010);
++	writel(0x10101010, host->base + MSDC_PAD_TUNE0);
```

Comment:
isn't tune0 handled already?

Reply:
MT762x only operates on register `MSDC_PAD_TUNE`. ref:
https://github.com/torvalds/linux/blob/256abd8e550ce977b728be79a74e1729438b4948/drivers/mmc/host/mtk-sd.c#L539

In our local driver. This register name `MSDC_DAT_RDDLY0`.
https://github.com/openwrt/openwrt/blob/e7599942af8e14541166234349e9e49278455ef6/target/linux/ramips/files/drivers/mmc/host/mtk-mmc/sd.c#L2107